# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLVertexArrayFake.cpp
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGLVertexArrayFake.cpp
--- a/dom/canvas/WebGLVertexArrayFake.cpp	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/canvas/WebGLVertexArrayFake.cpp	Wed Jan 08 22:19:16 2020 +0000
@@ -11,18 +11,18 @@
 namespace mozilla {
 
 WebGLVertexArrayFake::WebGLVertexArrayFake(WebGLContext* webgl)
-    : WebGLVertexArray(webgl, 0) {}
+    : WebGLVertexArray(webgl) {}
 
 void WebGLVertexArrayFake::BindVertexArray() {
   // Go through and re-bind all buffers and setup all
   // vertex attribute pointers
   gl::GLContext* gl = mContext->gl;
 
-  WebGLRefPtr<WebGLVertexArray> prevVertexArray = mContext->mBoundVertexArray;
+  RefPtr<WebGLVertexArray> prevVertexArray = mContext->mBoundVertexArray;
 
   mContext->mBoundVertexArray = this;
 
-  WebGLRefPtr<WebGLBuffer> prevBuffer = mContext->mBoundArrayBuffer;
+  RefPtr<WebGLBuffer> prevBuffer = mContext->mBoundArrayBuffer;
   mContext->BindBuffer(LOCAL_GL_ELEMENT_ARRAY_BUFFER, mElementArrayBuffer);
 
   size_t i = 0;